### PR TITLE
fix identifiers order on composite identifiers

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ∅
| License       | MIT
| Doc PR        | ∅

Actually the doctrine identifiers order is not necessary the same as the `propertyNameCollectionFactory->create($resourceClass)`. This should work better and don't rely on doctrine identifiers, but only on properties. 

Note that the order of the `propertyNameCollectionFactory->create` is the one of `getter/setters`, whereas doctrine identifiers order may be the one of annotation declaration.

### CODE REVIEW

**Comment by Simperfit** [comment]
LGTM

**Comment by dunglas** [comment]
Thanks @soyuka!

**Comment by teohhanhui** [comment]
This part of the code is pretty hard to follow lol...

**Comment by soyuka** [comment]
^^ why? I removed 12 lines, should be easier now \o/
